site/app/controllers/admin/PlagiarismController.php

```diff
@@ -533,161 +636,280 @@ public function saveNewPlagiarismConfiguration(string $new_or_edit, string $grad
-        assert($version_option == "active_version" || $version_option == "all_versions");
+                // This is a little ugly/repetitive but it can be frustrating for users to get nondescriptive errors
+                // so we try to make potential error cases a little more helpful
+                $error_message = "";
+                if (!isset($data["version"])) {
+                    $error_message = "Error: Invalid or missing version field";
+                }
+                elseif (!isset($data["regex"])) {
+                    $error_message = "Error: Invalid or missing regex field";
+                }
+                elseif (!isset($data["regex_dirs"])) {
+                    $error_message = "Error: Invalid or missing regex_dirs field";
+                }
+                elseif (!isset($data["language"])) {
+                    $error_message = "Error: Invalid or missing language field";
+                }
+                elseif (!isset($data["threshold"])) {
+                    $error_message = "Error: Invalid or missing threshold field";
+                }
+                elseif (!isset($data["sequence_length"])) {
+                    $error_message = "Error: Invalid or missing sequence_length field";
+                }
+                elseif (!isset($data["prior_term_gradeables"])) {
+                    $error_message = "Error: Invalid or missing prior_term_gradeables field";
+                }
+                elseif (!isset($data["ignore_submissions"])) {
+                    $error_message = "Error: Invalid or missing ignore_submissions field";
+                }
 
+                if ($error_message !== "") {
+                    throw new Exception($error_message);
+                }
 
-        // Regex ///////////////////////////////////////////////////////////////
-        // TODO: Can we find a way to validate the regex more thoroughly to tell the user their regex was invalid before feeding it to Lichen?
-        if (!isset($_POST["regex_dir"]) || !isset($_POST["regex_to_select_files"]) || str_contains($_POST["regex_to_select_files"], "..")) {
-            $this->core->addErrorMessage("Invalid regex form data.");
-            return new RedirectResponse($return_url);
+                $new_config = new PlagiarismConfig(
+                    $gradeable_id,
+                    $config_id,
+                    $data["version"],
+                    $data["regex"],
+                    in_array("submissions", $data["regex_dirs"]),
+                    in_array("results", $data["regex_dirs"]),
+                    in_array("checkout", $data["regex_dirs"]),
+                    $data["language"],
+                    $data["threshold"],
+                    $data["sequence_length"],
+                    $data["prior_term_gradeables"],
+                    $data["ignore_submissions"]
+                );
+                $em->persist($new_config);
+                $em->flush();
```

Comment:
The `->flush()` method should be ideally called as few times as possible, unless there's something specific why you need the DB changes to happen now, as opposed after batching all of them together. It's not clear to me why we couldn't just call this once at the end of the function.

Reply:
Isn't this the only `->flush()` method around?  I thought it would be best to put the `->flush()`near where the only other writing DB interactions occur.  If I'm not mistaken, this is the last (and only) place where writing DB operations occur.

Reply:
I would suggest moving the `flush()` here to be done after the if/else branching. Calling it having not entered either branch will be a no-op, but otherwise does we want, reducing the number of times we have this line by one. We also don't want to wrap the `flush` in a try/catch since if it fails, that would be a fatal error we want to abort on.